pynumaflow/function/server.py

```diff
@@ -218,102 +183,17 @@ def MapTFn(
-        )
-
-        # Save a reference to the result of this function, to avoid a
-        # task disappearing mid-execution.
-        self.background_tasks.add(response_task)
-        response_task.add_done_callback(lambda t: self.background_tasks.remove(t))
-
-        await response_task
-        results_futures = response_task.result()
-
-        for fut in results_futures:
-            await fut
-            yield udfunction_pb2.DatumResponseList(elements=fut.result())
-
-    async def __async_reduce_handler(self, interval_window, datum_iterator: AsyncIterable[Datum]):
-        callable_dict = {}
-        # iterate through all the values
-        async for d in datum_iterator:
-            keys = d.keys()
-            unified_key = DELIMITER.join(keys)
-            result = callable_dict.get(unified_key, None)
-
-            if not result:
-                niter = NonBlockingIterator()
-                riter = niter.read_iterator()
-                # schedule an async task for consumer
-                # returns a future that will give the results later.
-                task = asyncio.create_task(
-                    self.__invoke_reduce(keys, riter, Metadata(interval_window=interval_window))
-                )
-                # Save a reference to the result of this function, to avoid a
-                # task disappearing mid-execution.
-                self.background_tasks.add(task)
-                task.add_done_callback(lambda t: self.background_tasks.remove(t))
-                result = ReduceResult(task, niter, keys)
-
-                callable_dict[unified_key] = result
-
-            await result.iterator.put(d)
-
-        for unified_key in callable_dict:
-            await callable_dict[unified_key].iterator.put(STREAM_EOF)
-
-        tasks = []
-        for unified_key in callable_dict:
-            fut = callable_dict[unified_key].future
-            tasks.append(fut)
-
-        return tasks
-
-    async def __invoke_reduce(
-        self, keys: List[str], request_iterator: AsyncIterable[Datum], md: Metadata
-    ):
-        try:
-            msgs = await self.__reduce_handler(keys, request_iterator, md)
-        except Exception as err:
-            _LOGGER.critical("UDFError, re-raising the error: %r", err, exc_info=True)
-            raise err
-
-        datum_responses = []
-        for msg in msgs.items():
-            datum_responses.append(
-                udfunction_pb2.DatumResponse(keys=msg.keys, value=msg.value, tags=msg.tags)
-            )
-
-        return datum_responses
+    # def ReduceFn(
+    #         self,
+    #         request_iterator: AsyncIterable[udfunction_pb2.Datum],
+    #         context: NumaflowServicerContext,
+    # ) -> udfunction_pb2.DatumList:
+    #     """
+    #     Applies a reduce function to a datum stream.
+    #     The pascal case function name comes from the proto udfunction_pb2_grpc.py file.
+    #     """
+    #     _LOGGER.error("Reduce not supported on NEW SYNC --")
```

Comment:
I think we should either remove this, or just raise a `NotImplementedError`

Reply:
Done